### PR TITLE
Respecting regions

### PR DESCRIPTION
**Pull request by colins44**
this change helps the GET responses to be more region-specific. 

It also has changes to do with indexing data for each region, jobs for indexing data for each region have to be set up in jenkins.

### CODE REVIEW

File: src/scripts/indexOffers.ts [hunk 1]
```diff
@@ -135,6 +139,13 @@ export async function indexOffers(): Promise<any> {
           );
         }
         indexedOffers.push(offer.id_salesforce_external);
+        // using sorted sets incase we need to add more data and do lexographical
+        // indexing at a later date
```
Comment (sung-luxuryescapes) [inline src/scripts/indexOffers.ts:143]: nice we can do `zrangebylex`

File: src/models/destinationAlert.ts [hunk 2]
```diff
@@ -13,14 +13,15 @@ function getKey(value: any, keyType: string): string {
 }
 
 export async function getUserDestinationAlerts(
-  user: IUser
+  user: IUser,
+  region: any
```
Comment (sung-luxuryescapes) [inline src/models/destinationAlert.ts:17]: isn't it `string`?

File: src/models/popularDestination.ts [hunk 3]
```diff
@@ -9,9 +9,12 @@ const POPULAR_DESTINATION_USER = {
   herokuId: "popularDestination"
 };
 
-export async function getPopularDestinations(): Promise<IAlertObject[]> {
+export async function getPopularDestinations(
+  region: any
```
Comment (sung-luxuryescapes) [inline src/models/popularDestination.ts:13]: `string`?

File: src/scripts/indexOffers.ts (hunk 1 above)
Comment (colins44) [reply src/scripts/indexOffers.ts:143]: yup, that was trying to future proof a bit

File: src/models/destinationAlert.ts (hunk 2 above)
Comment (colins44) [reply src/models/destinationAlert.ts:17]: hi @sung yes it is, but query params are not a set type, they are one of either `string`, `array`, `object` or `number` so in this instance i use any in order to make my typescript compile.

Comment (colins44) [reply src/models/destinationAlert.ts:17]: @sung i did not know that you could do this in typescript:

```
const region: string = req.query.region as string || "au";
```

**Pull request merged**